crypto/signatures.go

```diff
@@ -69,8 +71,35 @@ func VerifyHash(data Hash, pk PublicKey, sig Signature) error {
 	return nil
 }
 
+// WriteSignedData writes a length-prefixed object followed by its signature.
+func WriteSignedData(w io.Writer, obj interface{}, sk SecretKey) error {
+	b := encoding.Marshal(obj)
+	sig, _ := SignHash(HashBytes(b), sk) // no error possible
+	return encoding.NewEncoder(w).EncodeAll(b, sig)
+}
```

Comment:
yeah, I think I agree. It would be technically possible to stream it -- you'd use a `MultiWriter` to write to both `w` and the hasher -- but the efficiency gain would be minimal unless the object is very large.